component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,394 @@
+						                                        accessToken);
+					}
+				} else {
+					throw new AuthenticationFailedException("Could not receive a valid access token from Pinterest");
+				}
+			} catch (OAuthSystemException e) {
+				throw new AuthenticationFailedException("Exception while building access token request", e);
+			} catch (ApplicationAuthenticatorException e) {
+				throw new AuthenticationFailedException("Exception while building the claim mapping", e);
+			}
+		} catch (OAuthProblemException e) {
+			throw new AuthenticationFailedException("Exception while getting the access token form the response", e);
+		}
+	}
+
+	/**
+	 * This method is to get the Pinterest user details.
+	 *
+	 * @param url         user info endpoint.
+	 * @param accessToken access token.
+	 * @return user info
+	 * @throws ApplicationAuthenticatorException
+	 */
+	private JSONObject fetchUserInfo(String url, String accessToken) throws ApplicationAuthenticatorException {
+		if (log.isDebugEnabled()) {
+			log.debug("Sending the request for getting the user info");
+		}
+		StringBuilder jsonResponseCollector = new StringBuilder();
+		BufferedReader bufferedReader = null;
+		HttpURLConnection httpConnection = null;
+		JSONObject jsonObj = null;
+		try {
+			URL obj = new URL(url + "?" + PinterestAuthenticatorConstants.ACCESS_TOKEN + "=" + accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setConnectTimeout(PinterestAuthenticatorConstants.CONNECTION_TIMEOUT_VALUE);
+				httpConnection.setReadTimeout(PinterestAuthenticatorConstants.READ_TIMEOUT_VALUE);
+				httpConnection.setRequestMethod(PinterestAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
+			}
+			String inputLine = bufferedReader.readLine();
+			while (inputLine != null) {
+				jsonResponseCollector.append(inputLine).append("\n");
+				inputLine = bufferedReader.readLine();
+			}
+			jsonObj = new JSONObject(jsonResponseCollector.toString());
+		} catch (MalformedURLException e) {
+			throw new ApplicationAuthenticatorException("MalformedURLException while generating the user info URL: "
+			                                            + url, e);
+		} catch (ProtocolException e) {
+			throw new ApplicationAuthenticatorException("ProtocolException while setting the request method: " +
+			                                            PinterestAuthenticatorConstants.HTTP_GET_METHOD +
+			                                            " for the URL: " + url, e);
+		} catch (IOException e) {
+			throw new ApplicationAuthenticatorException("IOException while reading the response from " + url, e);
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/1/commits/ddca573b64c6a6a1c7d9d8e270863e82ca9b77c9